tests/unit/components/sl-dialog-test.js

```diff
@@ -23,48 +46,140 @@ test( '"show" property defaults to false', function( assert ) {
+    );
+
+    assert.equal(
+        this.$( 'div.modal-dialog > div.modal-content > div.modal-header > span.modal-title' ).length,
+        1,
+        'Rendered component contains modal-title'
+    );
+
+    assert.equal(
+        this.$( 'div.modal-dialog > div.modal-content > div.modal-body' ).length,
+        1,
+        'Rendered component contains modal-body'
+    );
+
+    assert.equal(
+        this.$( 'div.modal-dialog > div.modal-content > div.modal-footer' ).length,
+        1,
+        'Rendered component contains modal-footer'
+    );
+
+    assert.equal(
+        this.$( 'div.modal-dialog > div.modal-content > div.modal-footer > button.btn-primary[data-dismiss="modal"]' ).length,
+        1,
+        'Rendered component contains modal dismiss button'
+    );
 });
 
 test( 'If "title" property is not populated, span with "modal-title" class is not rendered', function( assert ) {
-    this.render();
-
-    assert.equal( $('.modal-title').length, 0 );
+    assert.equal(
+        this.$( '.modal-title').length,
+        0,
+        'Rendered component does not contain modal-title'
+    );
 });
 
 test( '"title" is rendered as span with "modal-title" class if populated', function( assert ) {
-    var testTitle  = 'Test Title',
-        component  = this.subject({
-            title: testTitle
-        });
-
-    this.render();
+    var testTitle = 'Test Title',
+        component = this.subject({ title: testTitle });
+
+    assert.equal(
+        this.$( 'span.modal-title' ).length,
+        1,
+        'Rendered component contains modal-title'
+    );
+
+    assert.equal(
+        Ember.$.trim( $('.modal-title').text() ),
+        testTitle,
+        'Rendered modal-title contains expected text'
+    );
+});
 
-    assert.equal( $('span.modal-title').length, 1 );
-    assert.equal( $.trim( $('.modal-title').text() ), testTitle );
+test( 'The "show" property toggles the Bootstrap classes', function( assert ) {
+    var component = this.subject({ animated: false });
+
+    assert.strictEqual(
+        component.get( 'show' ),
+        false,
+        'Component is in hidden state initially'
+    );
+
+    assert.strictEqual(
+        this.$().is( ':visible' ),
+        false,
+        'Rendered component is hidden initially'
+    );
+
+    Ember.run( () => {
+        component.set( 'show', true );
+    });
+    assert.ok(
+        this.$().is( ':visible' ),
+        'Element is visible after first toggle'
+    );
+
+    Ember.run( () => {
+        component.set( 'show', false );
+    });
+    assert.strictEqual(
+        this.$().is( ':visible' ),
+        false,
+        'Element is hidden again after second toggle'
+    );
 });
 
-QUnit.skip( 'There are more tests to write', function( assert ) {
-    assert.expect(0);
+test( 'hideHandler properly handles hiding', function( assert ) {
+    var component = this.subject({ show: true });
 
-// remaining tests to write:
-// toggle
-// hideHandler
-// close
-// show
-// aria support
```

Comment:
toggle() is just an observer for 'show', which is tested for the relevant functionality.

Closing is Bootstrap behavior, and is unaffected by the Ember code, as long as the necessary attributes are present, which is again implemented in ModalMixin, and tested elsewhere.